pkg/controller/release/release_controller.go

```diff
@@ -317,7 +318,8 @@ func (c *Controller) syncOneReleaseHandler(key string) error {
 
 	if !releaseHasClusters(rel) {
 		shouldForce := false
-		if err := c.chooseClusters(rel, shouldForce); err != nil {
+		rel, err = c.chooseClusters(rel, shouldForce)
```

Comment:
Does any usage of `rel` up to this point require it to be deeply copied? Perhaps using `curRel` deep copy in this scope only as input of `chooseClusters` and keep referencing `curRel` everywhere can make it less ambiguous?

Another option would be `chooseClusters` perform a deep copy itself.